tests/ssg_test_suite/rule.py

```diff
@@ -296,11 +298,14 @@ def _get_rules_to_test(self):
             full_rule_id = OSCAP_RULE + short_rule_id
             if not self._rule_matches_rule_spec(short_rule_id):
                 continue
-            if not xml_operations.find_rule_in_benchmark(
-                    self.datastream, self.benchmark_id, full_rule_id):
-                logging.error(
-                    "Rule '{0}' isn't present in benchmark '{1}' in '{2}'"
-                    .format(full_rule_id, self.benchmark_id, self.datastream))
+            if full_rule_id not in all_rules_in_benchmark:
+                # This is an error only if the user specified the rules to be
+                # tested explicitly using command line arguments
+                if self.rule_spec:
```

Comment:
How about using `self.target_type` instead of `self.rule_spec`?

Also, how about inverting the conditionals?
If `self.target_type == rule ID`, then we check the presence of rule in Benchmark;
otherwise we just `continue`